__tests__/integration.test.js

```diff
@@ -257,6 +282,28 @@ describe('GET: api/users', () => {
     });
 });
 
+describe('GET: api/articles?topic', () => {
+    test('200: responds successfully with articles with specific topic', () => {
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toHaveLength(1)
+            body.forEach((article) => {
+                expect(article.topic).toBe("cats")
+            });
+        });
+    });
+    test('404: gives appropriate error when topic does not exist', () => {
+        return request(app)
+        .get('/api/articles?topic=banana')
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Not Found")
+        })
+    })
+});
+
```

Comment:
One really important test that is missing from this is a test to ensure we receive an empty array when given a topic that exists but has no associated articles. You will have to look at the data to see which topic this will be.